build/ci/clean-up-cloud-manager.sh

```diff
@@ -17,6 +17,11 @@
 set -euo pipefail
 
 # shellcheck disable=SC1091
-source project.sh
+if [[ -f project.sh ]]; then
+  source project.sh
+fi
 
-mongocli iam projects remove "$MCLI_PROJECT_ID"
+if [[ "${MUST_CLEANUP_CM-}" == "true" ]]; then
+  echo "cleaning up cloud manager project $MCLI_PROJECT_ID"
+  mongocli iam projects delete "$MCLI_PROJECT_ID" --force
+fi
```

Comment:
This command may fail depending on in-progress operations such as cluster or private endpoints deletions. Could you update this logic to attempt a number of times? We should have something similar to what we are currently doing with atlas -> see https://github.com/mongodb/mongodb-atlas-cli/blob/18cf627ffb68d191d76c87b860dd05c497647010/e2e/atlas/atlas_e2e_test_generator_test.go#L81